test/jasmine/tests/svg_text_utils_test.js

```diff
@@ -300,12 +300,36 @@ describe('svg+text utils', function() {
         it('allows one <sub> to span <br>s', function() {
             var node = mockTextSVGElement('SO<sub>4<br>44</sub>');
             expect(node.html()).toBe(
-                '<tspan class="line" dy="0em">SO​' +
+                '<tspan class="line" dy="0em">SO\u200b' +
                     '<tspan style="font-size:70%" dy="0.3em">4</tspan>' +
-                    '<tspan dy="-0.21em">​</tspan></tspan>' +
-                '<tspan class="line" dy="1.3em">​' +
+                    '<tspan dy="-0.21em">\u200b</tspan></tspan>' +
+                '<tspan class="line" dy="1.3em">\u200b' +
                     '<tspan style="font-size:70%" dy="0.3em">44</tspan>' +
-                    '<tspan dy="-0.21em">​</tspan></tspan>');
+                    '<tspan dy="-0.21em">\u200b</tspan></tspan>');
+        });
+
+        it('allows nested tags to break at <br>, eventually closed or not', function() {
+            var textCases = [
+                '<b><i><sup>many<br>lines<br>modified',
+                '<b><i><sup>many<br>lines<br>modified</sup></i></b>',
+                '<b><i><sup>many</sup><br><sup>lines</sup></i><br><i><sup>modified',
```

Comment:
This behavior *is* supported by regular HTML - and can be really useful to decouple styling from line breaks.

We tried weakly to support this previously but only allowed closing and reopening a single tag. The new version made it easy to handle arbitrary nesting.